code/modules/projectiles/updated_projectiles/guns/shotguns.dm

```diff
@@ -526,3 +531,94 @@ can cause issues with ammo types getting mixed up during the burst.
 	recoil_unwielded = CONFIG_GET(number/combat_define/high_recoil_value)
 	pump_delay = CONFIG_GET(number/combat_define/mhigh_fire_delay) * 2
 
+//------------------------------------------------------
+//A hacky bolt action rifle. in here for the "pump" or bolt working action.
+
+/obj/item/weapon/gun/shotgun/pump/bolt
+	name = "\improper mosin nagant rifle"
+	desc = "A mosin nagant rifle, even just looking at it you can feel the cosmoline already."
+	icon_state = "mosin"
+	item_state = "mosin" //thank you Alterist
+	fire_sound = 'sound/weapons/gun_sniper.ogg'
+	current_mag = /obj/item/ammo_magazine/internal/shotgun/pump/bolt
+	gun_skill_category = GUN_SKILL_RIFLES
+	type_of_casings = "cartridge"
+	pump_sound = 'sound/weapons/working_the_bolt.ogg'
+	attachable_allowed = list(
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/scope/mini,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/bayonet)
+	attachable_offset = list("muzzle_x" = 50, "muzzle_y" = 21,"rail_x" = 8, "rail_y" = 21, "under_x" = 37, "under_y" = 16, "stock_x" = 20, "stock_y" = 14)
+	starting_attachment_types = list(/obj/item/attachable/scope)
+
+/obj/item/weapon/gun/shotgun/pump/bolt/Initialize()
+	. = ..()
+	var/obj/item/attachable/S = new /obj/item/attachable/mosinbarrel(src)
```

Comment:
see #639 for how to do this in a way that isnt ass

Reply:
Apparently it's the `flags_attach_features = NOFLAGS` that can make an attachment unremovable. thus making this part uneeded.

Reply:
yep